http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -806,14 +808,17 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
   }
 
   private[service] def getClusterImages(
+    userEmail: WorkbenchEmail,
+    googleProject: GoogleProject,
     clusterRequest: ClusterRequest
   )(implicit ev: ApplicativeAsk[IO, TraceId]): IO[Set[ClusterImage]] =
     for {
       now <- IO(Instant.now)
       traceId <- ev.ask
+      petTokenOpt <- serviceAccountProvider.getAccessToken(userEmail, googleProject)
```

Comment:
This is cached under the hood so should not be a performance penalty.

Note this has to be the pet token as opposed to the user token, since the user token does not have the necessary scopes.

Reply:
I can run performance tests though..

Reply:
I ran a test and seemed to get good results:
```
now that's a nice test:
```
13:25:23 2020-01-10 13:25:23                                         544s elapsed
13:25:23 ---- Requests ------------------------------------------------------------------
13:25:23 > Global                                                   (OK=3400   KO=0     )
13:25:23 > notebooks:PUT:/api/cluster/v2/{googleProject}/{cluste... (OK=200    KO=0     )
13:25:23 > notebooks:GET:/api/cluster/{googleProject}/{clusterNa... (OK=200    KO=0     )
13:25:23 > notebooks:GET:/api/clusters/{googleProject}:list clus... (OK=2200   KO=0     )
13:25:23 > proxy:POST:/proxy/{googleProject}/{clusterName}/welde... (OK=200    KO=0     )
13:25:23 > proxy:POST:/proxy/{googleProject}/{clusterName}/welde... (OK=200    KO=0     )
13:25:23 > proxy:POST:/proxy/{googleProject}/{clusterName}/welde... (OK=200    KO=0     )
13:25:23 > notebooks:DELETE:/api/cluster/{googleProject}/{cluste... (OK=200    KO=0     )
13:25:23 
13:25:23 ---- LeoCreateMonitorDeleteSimulation_200_users_5_seconds_v1.0 -----------------
```